inc/datahandlers/user.php

```diff
@@ -324,14 +324,78 @@ function verify_website()
 	 */
 	function verify_icq()
 	{
+		global $mybb;
+
 		$icq = &$this->data['icq'];
 
+		if($mybb->settings['allowicqfield'] == '' || !is_member($mybb->settings['allowicqfield'], getuser($this->data['uid'])))
+		{
+			$icq = '';
```

Comment:
These three fields are still displayed in the ACP when editing an user. You either need to remove this from the data handler or hide them from all areas in the front end.

I'm unsure if disabling at the data handler level is the best approach. I'm also unsure if the settings to disable these 
fields make it clear that they will stop being editable from the ACP. @mybb/developers

Reply:
Well, just recall that this PR was created to address issue #4015, which is concerned with input _validation_, and I think that it _is_ appropriate as suggested in that issue and implemented in that PR that the validation in question occurs in the data handler. The issue of inhibiting the UI _display_ of disabled settings is a separate, though of course related, one, which I don't think this PR should address. I have no problem with a new issue and/or PR being created to address that distinct but related matter.

Reply:
I agree that this validation should be in the data handler, as suggested. What I'm against is for the data handler to reset and ignore values based on settings that were checked outside the data handler before. Before this patch, changes from the ACP worked just fine, but with this patch changes will be ignored, thus this issue should be taken care of here,

I would be fine with the ACP input fields being hidden just as in the forums (keeping the group permission check in the data handler), but then I would suggest to update the settings to be clear that disabling for anyone also means disabling ACP capabilities to bypass such setting for updating user data.

I tagged the team here because although I'm sure there is an issue raised with these changes, I'm unsure about how we would want to deal with it, if as I suggest or something different. We could keep the settings check outside the data handler as before for example, but for me that looks somehow uglier.

Reply:
OK. Understood. My preference would be for the solution you suggest. Will be interested in what the team thinks.

Reply:
Yes, it looks like the _Allow * Field To Usergroups_ settings are taken care of outside the datahandler: https://github.com/mybb/mybb/blob/a4225c8456ee73341e880ce043f7ef553433a069/usercp.php#L226-L257